src/display/editor/freetext.js

```diff
@@ -492,8 +492,54 @@ class FreeTextEditor extends AnnotationEditor {
       // This editor was created in using copy (ctrl+c).
       const [parentWidth, parentHeight] = this.parentDimensions;
       if (this.annotationElementId) {
-        const [tx] = this.getInitialTranslation();
-        this.setAt(baseX * parentWidth, baseY * parentHeight, tx, tx);
+        // This stuff is hard to test: if something is changed here, please
+        // test with the following PDF file:
+        //  - freetexts.pdf
+        //  - rotated_freetexts.pdf
+        // Only small variations between the original annotation and its editor
+        // are allowed.
```

Comment:
Is there no way that we could have e.g. a rudimentary integration-test for this?
E.g. something along these lines:
 - Load the PDF document from issue #16633
 - Get the coordinates of the relevant FreeText-annotation.
 - Enable editing.
 - Get the coordinates of the "new" Editor-element.
 - Compare the two sets of coordinates, and ensure that their difference is "small".

Reply:
I already tried...
For the foxit annotation in freetexts.pdf:
 - without the patch (temp0 is the annotation and temp1 the editor):
   ![image](https://github.com/mozilla/pdf.js/assets/5641725/706c56cf-bf11-4b1e-a69c-3a180d358684)
 - with the patch:
   ![image](https://github.com/mozilla/pdf.js/assets/5641725/48a9720a-a360-486b-a09f-3ed549b4f17e)

This patch takes into account the padding added by the editor when generating the appearance so I don't really see how I can test that... The only solution I can imagine is to make a screenshot and then compare it with a reference but we've nothing to do that.

Reply:
I really don't like having to test manually something.... I'm really lazy but I way prefer spending one hour in writing a test instead of having to remember that I need to test something manually and then test the stuff in trying to remember the different scenarios.
So I think we should screenshot the page, put it somewhere and then compare it with a ref we've.
Do you think it's feasible ? Can you imagine something easier ?

Reply:
Taking the document from the issue, i.e. https://github.com/mozilla/pdf.js/files/11933959/sample.pdf, and modifying it as follows might help with writing an integration-test?
 - Changing the /MediaBox entries to e.g. `[-75 -75 600 850]`.
 - Removing *all* /BleedBox, /CropBox, and /TrimBox entries.

With these changes the differences between the annotationLayer and the annotationEditorLayer should be big enough (on `master`) that we could just compare the coordinates and ensure that they at least overlap.
Maybe I'm overlooking or oversimplifying a bunch of things, but would this be a possible solution?

Reply:
With the patch I get `{ x: 377.433349609375, y: 335.3666687011719, ...` for the annotation and `{ x: 372.1000061035156, y: 330.0333251953125,...} for the editor.
So we could that we've a tolerance of 5px but if we've a diff of 4px in the opposite direction then the test will be ok and rendering will be wrong.
And I'm wondering if these differences could be different with another screen resolution.
That said, for this specific case, the difference corresponds to the value returned by:
https://github.com/mozilla/pdf.js/blob/master/src/display/editor/freetext.js#L217